app/controllers/architectures_controller.rb

```diff
@@ -1,11 +1,27 @@
 class ArchitecturesController < ApplicationController
+  include ApplicationHelper
   include Foreman::Controller::AutoCompleteSearch
   include Foreman::Controller::Parameters::Architecture
 
   before_action :find_resource, :only => [:edit, :update, :destroy]
 
   def index
     @architectures = resource_base_search_and_page.includes(:operatingsystems)
+    respond_to do |format|
+      format.html do
+        render "index"
+      end
+      format.json do
+        render json: {
+          rows: rows(@architectures),
+          columns: columns,
+          item_count: @architectures.count,
+          header: _("Architectures"),
+          title: _("Foreman Architectures"), # optional, sets the browser tab title
```

Comment:
I don't think Foreman Architectures is right. Today we also just have Architectures and that's the right thing IMHO. That also makes me think: in a controller we already have the resource type. Can't we use `model_of_controller` to automatically determine the header and title? That again would reduce the duplication.

Perhaps @timogoebel already knows the right methods to call for that.

Reply:
it was an example of how some other plugin could override it,
+1 for the reuse, although `model_of_controller` won't be translated well if needed?